### PR TITLE
Refactoring of exceptions

### PR DESCRIPTION
**Pull request by TamoshaytisV**
I think it's a good idea to have base video xblock exceptions to distinguish them from other bad code.

@z4y4ts @dorosh please review

### CODE REVIEW

File: video_xblock/backends/base.py [hunk 1]
```diff
@@ -307,8 +303,8 @@ def get_transcript_language_parameters(lang_code):
         lang_code = lang_code[0:2]
         # Check on consistency with the pre-configured ALL_LANGUAGES
         if lang_code not in [language[0] for language in settings.ALL_LANGUAGES]:
-            raise Exception('Not all the languages of transcripts fetched from video platform are '
-                            'consistent with the pre-configured ALL_LANGUAGES')
+            raise VideoXBlockException(_('Not all the languages of transcripts fetched from video platform are '
```
Comment (z4y4ts) [inline video_xblock/backends/base.py]: Please follow [Open edX JavaScript code style guide](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html#breaking-long-lines).

File: video_xblock/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+from .utils import ugettext as _
+
+
+class VideoXBlockException(Exception):
+    """
+    Base class for video xblock exceptions.
+    Subclasses should provide `.default_detail` property.
+    """
+    default_detail = _('An exception occurred.')
```
Comment (z4y4ts) [inline video_xblock/exceptions.py]: `default_msg` and `msg` (or `message`) feels more appropriate here.

File: video_xblock/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    def __str__(self):
+        return self.detail
+
+
+class ApiClientError(VideoXBlockException):
+    """
+    Base API client exception.
+    """
+    default_detail = _('API error occurred.')
```
Comment (dorosh) [inline video_xblock/exceptions.py]: add last blank string

**Comment by TamoshaytisV** [comment]
@OPersian please also review

File: video_xblock/exceptions.py (hunk 3 above)
Comment (z4y4ts) [inline video_xblock/exceptions.py]: @dorosh it's there. Otherwise, GitHub would show something like `↵∅`

Comment (TamoshaytisV) [inline video_xblock/exceptions.py]: it was there :)

File: video_xblock/exceptions.py (hunk 2 above)
Comment (TamoshaytisV) [inline video_xblock/exceptions.py]: changed

File: video_xblock/backends/base.py (hunk 1 above)
Comment (TamoshaytisV) [inline video_xblock/backends/base.py]: fixed

**Comment by TamoshaytisV** [comment]
@z4y4ts please continue review

**Comment by z4y4ts** [comment]
Squash and merge.

**Pull request merged**